app/src/main/java/org/simple/clinic/scanid/ScanSimpleIdEffectHandler.kt

```diff
@@ -67,7 +67,13 @@ class ScanSimpleIdEffectHandler @AssistedInject constructor(
     return when (country.isoCountryCode) {
       Country.INDIA -> {
         val payload = qrCodeJsonParser.parseQRCodeJson(effect.text)
-        ScannedQRCodeJsonParsed(payload?.toPatientPrefillInfo(), payload?.healthIdNumber)
+        val healthIdNumber = payload?.healthIdNumber
+
+        if (!healthIdNumber.isNullOrBlank() && healthIdNumber.length < 14) {
```

Comment:
These kinds of checks should be part of our business logic, so we should ideally have them in `Update` and then decide what to do.

Reply:
Here we are referencing a magic number `14`. Ideally, we should have a variable that specifies this number so that we don't lose the context of why it should be less than 14 characters.

Reply:
Also, what should happen when the health ID number has more than 14 characters?